sqlmesh/integrations/dlt.py

```diff
@@ -0,0 +1,169 @@
+import typing as t
+import click
+from pydantic import ValidationError
+from sqlglot import exp
+from sqlmesh.core.config.connection import parse_connection_config
+
+
+def generate_dlt_models_and_settings(
+    pipeline_name: str, dialect: str
+) -> t.Tuple[t.Set[t.Tuple[str, str]], str]:
+    """This function attaches to a DLT pipeline and retrieves the connection configs and
+    SQLMesh models based on the tables present in the pipeline's default schema.
+    """
+
+    import dlt
+    from dlt.common.schema.utils import has_table_seen_data, is_complete_column
+    from dlt.pipeline.exceptions import CannotRestorePipelineException
+
+    try:
+        pipeline = dlt.attach(pipeline_name=pipeline_name)
+    except CannotRestorePipelineException:
+        raise click.ClickException("Could not attach to pipeline {pipeline_name}")
+
+    schema = pipeline.default_schema
+    dataset = pipeline.dataset_name
+
+    client = pipeline._sql_job_client(pipeline.default_schema)
+    config = client.config
+    credentials = config.credentials
+    db_type = pipeline.destination.to_name(pipeline.destination)
+    configs = {
+        key: value
+        for key in dir(credentials)
+        if not key.startswith("_")
+        and not callable(value := getattr(credentials, key))
+        and value is not None
+    }
+
+    dlt_tables = {
+        name: table
+        for name, table in schema.tables.items()
+        if (has_table_seen_data(table) and not name.startswith(schema._dlt_tables_prefix))
+        or name == schema.loads_table_name
+    }
+
+    sqlmesh_models = set()
+    sqlmesh_schema_name = f"{dataset}_sqlmesh"
+    for table_name, table in dlt_tables.items():
+        dlt_columns = {
+            c["name"]: exp.DataType.build(str(c["data_type"]), dialect=dialect)
+            for c in filter(is_complete_column, table["columns"].values())
+            if c.get("name") and c.get("data_type")
+        }
+        primary_key = [
+            str(c["name"])
+            for c in filter(is_complete_column, table["columns"].values())
+            if c.get("primary_key") and c.get("name")
+        ]
+        model_def_columns = format_columns(dlt_columns, dialect)
+        select_columns = format_columns_for_select(dlt_columns)
+        grain = format_grain(primary_key)
+        incremental_model_name = f"{sqlmesh_schema_name}.incremental_{table_name}"
+        full_model_name = f"{sqlmesh_schema_name}.full_{table_name}"
+
+        incremental_model_sql = generate_incremental_model(
```

Comment:
Why do we produce both incremental and full models at the same time?

Reply:
We don't have to. My thinking was to create a project similar to the example project and from each dlt table to create an equivalent incremental and full model. So from the full model the user can remove the columns they don't need or remove them according to their use case

But we can simply generate one type of model that loads from the dlt tables and leave it to the user if they want then to have other models to source from them

Reply:
we should just generate one, what does their dbt generator do?

Reply:
the dbt generator creates staging and dimensional models under corresponding staging and marts directories for each dlt table. but I suppose yes we don't have to follow that logic, so I can modify the code and simply generate one type of model. and in that case should I keep the incremental kind to source from the dlt table?

Reply:
yes, i think it should be incremental, but what does it mean creates staging and dimension, and is that analogous to you do both incremental and full?

Reply:
no the staging and dimension in this case is for dbt to do the incremental processing but it isn't analogous. I simply followed the example project  that we have with both incremental and full kinds, so yes I can remove the full models